src/lib/matchers.rs

```diff
@@ -121,6 +121,18 @@ impl TemporaryMappingStore {
     pub fn get_to(&self, from: &NodeId) -> Option<NodeId> {
         self.from.get(from).and_then(|x| Some(*x))
     }
+
+    /// Test whether `n1` is mapped to `n2` in this store.
+    pub fn is_mapped(&self, from: &NodeId, to: &NodeId) -> bool {
+        if !self.contains_from(from) {
+            return false;
+        }
+        let mapped_to = self.get_to(from);
+        if mapped_to.is_none() {
```

Comment:
This feels like it might be neater as something like:

```rust
mapped_to.map_or(false, |x| x == to)
```
?

(same with the similar statement below)